packages/sdk/src/evm/common/plugin.ts

```diff
@@ -105,6 +110,61 @@ async function getPluginABI(
   ).map((metadata) => metadata.abi);
 }
 
+export async function getPluginNames(
+  publishedMetadataUri: string,
+  storage: ThirdwebStorage,
+): Promise<string[]> {
+  const extendedMetadata = await fetchExtendedReleaseMetadata(
+    publishedMetadataUri,
+    storage,
+  );
+
+  let contractNames: string[] = [];
+
+  if (extendedMetadata.factoryDeploymentData?.implementationAddresses) {
+    const implementationsAddresses = Object.entries(
+      extendedMetadata.factoryDeploymentData.implementationAddresses,
+    );
+
+    try {
+      for (const [network, implementation] of implementationsAddresses) {
+        if (implementation !== "") {
+          const provider = getChainProvider(parseInt(network), {});
+          const contract = new ContractWrapper(
+            provider,
+            implementation,
+            getAllPluginsAbi,
+            {},
+          );
+
+          const pluginMap = await contract.call("getAllPlugins");
```

Comment:
shouldn't assume that the caller of this function will always pass a router contract. this function should be doing the checks for what type of contract we're dealing with, and return [] if not a router contract